xoto3/dynamodb/exceptions.py

```diff
@@ -1,27 +1,75 @@
 """Exceptions for our Dynamo usage"""
+from typing import Optional, TypeVar, Type, Dict, Tuple
 import botocore.exceptions
 
+from .types import ItemKey
+
 
 class DynamoDbException(Exception):
     """Wrapping error responses from Dynamo DB"""
 
     pass
 
 
-class AlreadyExistsException(DynamoDbException):
+class DynamoDbItemException(DynamoDbException):
+    def __init__(self, msg: str, *, key: Optional[ItemKey] = None, table_name: str = "", **kwargs):
+        self.__dict__.update(kwargs)
+        self.key = key
+        self.table_name = table_name
+        super().__init__(msg)
+
+
+class AlreadyExistsException(DynamoDbItemException):
+    """Deprecated - prefer ItemAlreadyExistsException"""
+
+    pass
```

Comment:
FYI...pass is not required if a docstring is used...

Reply:
yeah, it was there before i added the comment and then i didn't think to remove it.